FunctionalProgramming/Recursion/PrefixCompression.fsx

```diff
@@ -1,10 +1,32 @@
 open System
 
-let findMatchingPrefix' (input1:string) (input2:string) =
-    input2
-    |> Seq.zip input1
-    |> Seq.takeWhile(fun (a, b) -> a = b)
-    |> Seq.map(fun (a, b) -> a |> string)
+let inline equal (a, b) = a = b
+// `inline` requests inline compilation, that means everywhere `equal` is
+// called, the actual code of the function will appear in the IL instead of
+// a jump.
+// It is only a request, though; as far as I know, the compiler *may*
+// decide against complying with it (presumably for larger functions). 
+
```

Comment:
The default is definitely to *not* use it. It is purely a performance optimization and should only be used for small functions that are called *a lot*, for example in tight loops. Most/all(?) operators in `FSharp.Core` are declared as `inline`, and that makes sense, because you wouldn't want to incur a jump (at the IL level) for every mathematical operation or pipe call. One of the biggest caveats of using it is that it severely complicates debugging - as mentioned, inlining a function causes its code to be placed at every call site in the IL, so when you run over that code in the debugger, your compiled code looks completely different from your F# source in that place, and the IDE won't be able to tell you where you really are in your code. (All that of course only if the compiler agreed to actually inline it in the first place.)